runtime/parser2/type.go

```diff
@@ -430,26 +450,62 @@ func defineRestrictedOrDictionaryType() {
 		},
 	)
 
-	// For the left denotation we definitely know it is a restricted type
+	// For the left denotation we need a meta left denotation:
+	// We need to look ahead and check if the brace is followed by whitespace or not.
+	// In case there is a space, the type is *not* considered a restricted type.
+	// This handles the ambiguous case where a function return type's open brace
+	// may either be a restricted type (if there is no whitespace)
+	// or the start of the function body (if there is whitespace).
+	//
+	// In the fu
 
-	setTypeLeftBindingPower(lexer.TokenBraceOpen, typeLeftBindingPowerRestriction)
-	setTypeLeftDenotation(
+	//setTypeLeftBindingPower(lexer.TokenBraceOpen, typeLeftBindingPowerRestriction)
+	setTypeMetaLeftDenotation(
```

Comment:
Looks like some leftover comments on these lines

Reply:
Good catch, removed in 21fa7ae!